Feature: highlights scene items

emptybraces opened the pull request:
Add color to the left side of the scene item to highlight it.

When multiple scenes are open or when many headers are used, it was frustrating because the scene boundaries were hard to tell apart at a glance.. I'm not sure if this is the best solution, but it has definitely improved.
(It might have been an inappropriate pull request since it seems like a personal feature)
![image](https://github.com/user-attachments/assets/77d76551-096d-4bbe-8f99-b31f54ea12dc)

WooshiiDev commented:
I like this! It's very light too, however I feel this is very limited right now.

I feel this could be extended slightly:
 - Make this a setting to toggle
 - Add a setting for the colour used
 - A setting for line thickness would be nice too

emptybraces commented:
![image](https://github.com/user-attachments/assets/d779a072-a4b3-4a81-b534-1fec6cbb4ffc)
The config file already has the fields set up.

On the left side, there are two option toggle icons related to the scene view.
To make these icons visible, the only option is to fade them using alpha.
It might be possible to redraw the icons using the existing style feature with prefixes, but that seems a bit complicated.

emptybraces commented:
I added thickness field. Is this what you meant?

https://github.com/user-attachments/assets/2d7dd7af-27b5-4e5d-9dca-a820d4c652de

emptybraces commented:
Hi! 
This feature wasn’t working in Unity 6, so I looked into it and fixed an incorrect scene ID comparison. Also adjusted the appearance as you mentioned, is this okay?

https://github.com/user-attachments/assets/2d199567-0902-4967-b162-64301d188de4

WooshiiDev commented:
Hi empty!
Thanks for getting those changes in ❤️ I appreciate every small thing you've done for this.

Final few small changes and I'm happy merging this in immediatey:
- I personally feel the default colour should be the same as the header for scenes in the hierarchy which seems to be `292929`. Would you be able to change it to that? 🙏 The alpha is fine though as the icons need to be shown
- Should we also consider the icons that this will draw over? Do these need redrawn again? Redrawing stuff like with the two-tone feature isn't the most optimal though. If you have other ideas throw them here. 

Also as a side note, I know you have some fixes for excluding from builds and the likes. If any of them are still valid, I'm happy to merge PRs for them, I'm going to make a new version with this addition and some small changes before I go onto a big overhaul of how I cache a lot of stuff (performance related because its been left too long).

emptybraces commented:
Thank you for checking!

I changed the default color. Background color in dark theme is #292929, but it’s different in light theme. But it seems fine to me.
Since I’m just drawing a rectangle afterward, it ends up covering the two icons that show up on mouse hover. So the alpha setting is required. It can’t really be avoided 😅

> Also as a side note, I know you have some fixes for excluding from builds and the likes. If any of them are still valid, I'm happy to merge PRs for them,

Yeah, that’s implemented it in my fork.😎 It ended up being pretty personal and a bit rough, so when I have some time, I’ll clean up the appearance and the code and then submit a PR.

WooshiiDev commented:
Not a problem! Thank you for this, time to merge! I'll get this in a new release soon ❤️
